sqlite.go

```diff
@@ -490,6 +490,11 @@ func (s *stmt) ExecContext(ctx context.Context, args []driver.NamedValue) (drive
 	if err := s.bindAll(args); err != nil {
 		return nil, s.reserr("Stmt.Exec(Bind)", err)
 	}
+	if d, ok := ctx.Deadline(); ok && ctx.Value(queryTimeoutKey{}) != nil {
```

Comment:
This is basically what I was thinking, but you can handle general context cancelation here, not just deadlines, by using a goroutine and selecting on ctx.Done

Reply:
I did this so we wouldn't need another goroutine unless the cancellation actually fires.

Reply:
If `time.AfterFunc` is really that efficient, maybe it can always be on?

For most queries most of the time, spinning up a goroutine during the query is totally fine. It's just in our server full of hot paths we can't do it. So the idea of having an opt-in way to make contexts fully-functional seems very reasonable to me.

Reply:
AfterFunc still starts a goroutine when the timer ends, but in the common case where a query doesn't exceed its deadline it's just the timer entry. But if you don't think a goroutine for all queries is a concern that's just as easy to wire up. I'll do that.

Reply:
I pushed a separate commit to watch the context for any termination as you suggested.